codebase2/codebase-sqlite/U/Codebase/Sqlite/Queries.hs

```diff
@@ -645,31 +646,60 @@ addToDependentsIndex dependency dependent = execute sql (dependency :. dependent
     ON CONFLICT DO NOTHING
   |]
 
+-- | Get non-self, user-defined dependents of a dependency.
 getDependentsForDependency :: DB m => Reference.Reference -> m [Reference.Id]
-getDependentsForDependency dependency = query sql dependency where sql = [here|
-  SELECT dependent_object_id, dependent_component_index
-  FROM dependents_index
-  WHERE dependency_builtin IS ?
-    AND dependency_object_id IS ?
-    AND dependency_component_index IS ?
-|]
-
+getDependentsForDependency dependency =
+  filter isNotSelfReference <$> query sql dependency
```

Comment:
Might need to expand the `dependency` into a tuple to do so, but wouldn't it be better to filter out self-references in the query itself?

Reply:
It would be better, but I don't think it matters much. I was planning to make more efficient SQL in the next step of generalizing the query, so we don't have to rewrite it in place twice.